task-5/src/api/createUser.api.ts

```diff
@@ -0,0 +1,70 @@
+import { randomUUID } from "crypto";
+import { users_db } from "../db-mock/db";
+import { STATUS_CODES } from "../constants";
+import { LINKS_PATTERNS_MAP, LINKS_USER_ID_TEMPLATE } from "../config";
+import omitFields from "./helpers/omitFields";
+import generateResponse from "./helpers/generateResponse";
+
+import { IApiHandler, IUser, IUserDataRequest } from "../types";
+
+const isUserDataRequest = (userData: any): userData is IUserDataRequest =>
+  typeof userData?.name === "string" && typeof userData?.email === "string";
+
+const createNewUser = (userDataRequest: IUserDataRequest): IUser => ({
+  id: randomUUID(),
+  hobbies: [],
+  name: userDataRequest.name,
+  email: userDataRequest.email,
+});
+
+const createUser: IApiHandler = (req, res) => {
+  let requestBody = "";
+
+  req.on("data", (chunk) => {
+    requestBody += chunk;
+  });
+
+  req.on("end", () => {
+    const requestData = JSON.parse(requestBody);
+    const isValid = isUserDataRequest(requestData);
+
+    if (isValid) {
+      const newUser = createNewUser(requestData);
+      users_db.push(newUser);
```

Comment:
It's not required to do this as part of this task, but in general, you can add a separate service that will act as a layer between business requirements and the implementation of the database. This way, when changing the source, it will be easier to replace one database with another

Reply:
As discussed internally, Layered Architecture will be the part of the [next task](https://d17btkcdsmqrmh.cloudfront.net/node-gmp/docs/express-layered-architecture/layered_architecture).

Anyway, will do :)

Reply:
Done